DAQ/src/CaloRecoFromFragments_module.cc

```diff
@@ -231,26 +234,26 @@ void art::CaloRecoFromFragments::analyze_calorimeter_(
         // IMPORTANT NOTE: we don't have a final
         // mapping yet so for the moment, the BoardID field (described in docdb 4914) is just a
         // placeholder. Because we still need to know which crystal a hit belongs to, we are
-        // temporarily storing the 4-bit apdID and 12-bit crystalID in the Reserved DIRAC A slot.
+        // temporarily storing the 4-bit sipmID and 12-bit crystalID in the Reserved DIRAC A slot.
         // Also, note that until we have an actual map, channel index does not actually correspond
         // to the physical readout channel on a ROC.
-        uint16_t crystalID = hits[hitIdx].first.DIRACB & 0x0FFF;
-        uint16_t apdID = hits[hitIdx].first.DIRACB >> 12;
+        uint16_t crystalID = caloDAQUtil_.getCrystalID(hits[hitIdx].first); //hits[hitIdx].first.DIRACB & 0x0FFF;
+        uint16_t sipmID    = caloDAQUtil_.getSiPMID   (hits[hitIdx].first); //hits[hitIdx].first.DIRACB >> 12;
 
         // FIXME: Can we match vector types here?
         std::vector<int> caloHits;
 	caloHits.reserve(hits[hitIdx].second.size());
         std::copy(hits[hitIdx].second.begin(), hits[hitIdx].second.end(), std::back_inserter(caloHits));
 
-        calo_digis->emplace_back((crystalID * 2 + apdID), hits[hitIdx].first.Time, caloHits,
+        calo_digis->emplace_back((sipmID), hits[hitIdx].first.Time, caloHits,
                                  hits[hitIdx].first.IndexOfMaxDigitizerSample);
 
         if (diagLevel_ > 1) {
           // Until we have the final mapping, the BoardID is just a placeholder
           // adc_t BoardId    = cc.DBC_BoardID(pos,channelIdx);
 
           std::cout << "Crystal ID: " << (int)crystalID << std::endl;
-          std::cout << "APD ID: " << (int)apdID << std::endl;
+          std::cout << "SiPM ID: " << (int)sipmID << std::endl;
```

Comment:
Why do you need this cast?  I don't think that it does anything?  

In cases where you do need a cast, please use c++ style casts.

Reply:
done.

Reply:
There are many others of these in this file.  Please fix them all and I will merge right after.

Reply:
those I left are needed because they are uint8/16 and when I was printing them with cout was not readable.

Reply:
OK.  Merging now.